test/sql/types/timestamp/test_timestamp_to_tz_cast.test

```diff
@@ -0,0 +1,43 @@
+# name: test/sql/types/timestamp/test_timestamp_to_tz_cast.test
+# description: Test casting to TIMESTAMP WITH TIME ZONE from any unit
+# group: [timestamp]
+
+require icu
+
+# This breaks the 'verification_stats' check
+#statement ok
+#PRAGMA enable_verification
+
+# Set up some timestamp data
+statement ok
+create table ts_tbl as select '1990/12/21'::TIMESTAMP as ts;
+
+statement ok
+SET TimeZone='UTC'
```

Comment:
Add `SET Calendar='gregorian'` - sometime people are in locales where the default calendar is not (e.g., China).